src/drivers/Akamai.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+namespace ostark\upper\drivers;
+
+use GuzzleHttp\Exception\BadResponseException;
+use ostark\upper\exceptions\AkamaiApiException;
+
+/**
+ * Class Akamai Driver
+ *
+ * @package ostark\upper\drivers
+ *
+ */
+class Akamai extends AbstractPurger implements CachePurgeInterface
+{
+    /**
+     * @var string
+     */
+    public $host;
+
+    /**
+     * @var string
+     */
+    public $clientToken;
+
+    /**
+     * @var string
+     */
+    public $clientSecret;
+
+    /**
+     * @var string
+     */
+    public $accessToken;
+
+    /**
+     * @var string
+     */
+    public $maxSize;
+
+    /**
+     * Purge cache by tag
+     *
+     * @param string $tag
+     *
+     * @return bool
+     */
+    public function purgeTag(string $tag)
+    {
+        if ($this->useLocalTags) {
+            return $this->purgeUrlsByTag($tag);
+        }
+
+        $this->sendRequest('production', 'POST', 'tag', $tag);
+        $this->sendRequest('staging', 'POST', 'tag', $tag);
+
+        return true;
+    }
+
+    /**
+     * Purge cache by urls
+     *
+     * @param array $urls
+     *
+     * @return bool
+     */
+    public function purgeUrls(array $urls)
+    {
+        foreach ($urls as $url) {
+            if (!$this->sendRequest('production', 'POST', 'url', getenv('DEFAULT_SITE_URL') . $url)) {
+                return false;
+            }
+            if (!$this->sendRequest('staging', 'POST', 'url', getenv('DEFAULT_SITE_URL') . $url)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+
+    /**
+     * Purge entire cache
+     *
+     * @return bool
+     */
+    public function purgeAll()
+    {
+        // TODO: Purge all in Akamai
+        return true;
```

Comment:
is there no endpoint?

Reply:
Unfortunately, no. Only by tag, cp code or url. CP Code could work as a purge all, but only if you set it up specifically.

Reply:
I had a quick look at the docs. cp code should work.